### ISSUE TITLE
Kodi 18.5 crashes with kodi-vfs-rar installed (not even enabled)

### ISSUE THREAD

**Issue by boba23** [xbmc/vfs.rar#53]
Hi spiff,

I think since your version bump to 2.0.8 Kodi is crashing on me when trying to start any kind of video. Did some debugging, though Kodi log wasn't really helping, but what I get is, after trying to start a video, a few seconds later Kodi restarts itself and you see this in my Lubuntu 18.04.3 syslog:

kernel: [   22.351484] kodi-x11[1043]: segfault at 8 ip 00007fc1692e9c44 sp 00007ffe8040b458 error 4 in vfs.rar.so.2.0.8[7fc1692a0000+71000]

uninstalling kodi-vfs-rar fixes this. Never had any issues with it before. I now installed libarchive, which works, but is slower at opening rar archives when trying to start a movie.

Anyone else able to reproduce this?

**Comment by AlwinEsch** [comment]
Some Things:
- Which was your Version before, which runs? In 2.0.8 is mostly nothing changed in Code.
- Have you created by self or downloaded via `sudo apt install ...`?
- I think you have the second way via apt install, install also `kodi-vfs-rar-dbg`
- Make a debug run with:
  - Start Kodi
  - On Command line start `sudo gdb --pid=$(pidof kodi-x11)`
  - On gdb's command line enter a `c` character to continue Kodi run
  - Then reproduce the crash in RAR file
  - gdb then stops the run and message a crash
  - Inside them enter `bt`, which make a backtrace about crash
  - Then you can exit them with `exit`
  - Give us the given backtrace which looks e.g.
```
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
0x00007fe4293ce670 in __GI___nanosleep (requested_time=requested_time@entry=0x7fffc885e4f0, remaining=remaining@entry=0x0) at ../sysdeps/unix/sysv/linux/nanosleep.c:28
28      ../sysdeps/unix/sysv/linux/nanosleep.c: Datei oder Verzeichnis nicht gefunden.
(gdb) Quit
(gdb) bt
#0  0x00007fe4293ce670 in __GI___nanosleep (requested_time=requested_time@entry=0x7fffc885e4f0, remaining=remaining@entry=0x0) at ../sysdeps/unix/sysv/linux/nanosleep.c:28
#1  0x00007fe429401894 in usleep (useconds=<optimized out>) at ../sysdeps/posix/usleep.c:32
#2  0x0000558ffb3e20e9 in Sleep(unsigned int) (dwMilliSeconds=40) at /home/alwin/Development/Kodi/API/kodi-Matrix/xbmc/platform/posix/XTimeUtils.cpp:44
#3  0x0000558ffb3b3460 in CRenderSystemGL::PresentRender(bool, bool) (this=0x558ffec86eb8, rendered=false, videoLayer=false)
    at /home/alwin/Development/Kodi/API/kodi-Matrix/xbmc/rendering/gl/RenderSystemGL.cpp:290
#4  0x0000558ffb75c474 in CGraphicContext::Flip(bool, bool) (this=0x558ffec87010, rendered=false, videoLayer=false)
    at /home/alwin/Development/Kodi/API/kodi-Matrix/xbmc/windowing/GraphicContext.cpp:988
#5  0x0000558ffbca8d41 in CApplication::Render() (this=0x558ffebc4740) at /home/alwin/Development/Kodi/API/kodi-Matrix/xbmc/Application.cpp:1556
#6  0x0000558ffbd8669b in CXBApplicationEx::Run(CAppParamParser const&) (this=0x558ffebc4740, params=...) at /home/alwin/Development/Kodi/API/kodi-Matrix/xbmc/XBApplicationEx.cpp:73
#7  0x0000558ffb8d2dab in XBMC_Run(bool, CAppParamParser const&) (renderGUI=true, params=...) at /home/alwin/Development/Kodi/API/kodi-Matrix/xbmc/platform/xbmc.cpp:72
#8  0x0000558ffb1c2e4a in main(int, char**) (argc=1, argv=0x7fffc885e868) at /home/alwin/Development/Kodi/API/kodi-Matrix/xbmc/platform/posix/main.cpp:77
  ```

With:
```
kernel: [ 22.351484] kodi-x11[1043]: segfault at 8 ip 00007fc1692e9c44 sp 00007ffe8040b458 error 4 in vfs.rar.so.2.0.8[7fc1692a0000+71000]
```
is not direct visible which place exact makes the crash.

**Comment by boba23** [comment]
Hey Alwin,

many thanks for your extensive answer, but the issue seems to have been on my side. I did start over with a clean user directory before you answered and with that, vfs.rar works absolutely fine. So it must have been something with my setup. Though I still don't know what happened. The issue I described started from one day to the other. Before vfs.rar was working fine. 

Anyway, thanks again for your help.

boba
